Add app URL in start command

gtsiolis opened the pull request:
<!-- Please refer to the contribution guidelines before raising a PR: https://github.com/localstack/localstack/blob/master/docs/CONTRIBUTING.md -->

<!-- Why am I raising this PR? Add context such as related issues, PRs, or documentation. -->
## Motivation

See **[relevant discussion](https://localstack-cloud.slack.com/archives/C06JVGFM1NH/p1733496727046419?thread_ts=1733342778.435109&cid=C06JVGFM1NH)** where initially the idea of linking to the web app was mentioned.

<!-- What changes does this PR make? How does LocalStack behave differently now? -->
## Changes

| BEFORE | AFTER |
|--------|--------|
| <img width="638" alt="Screenshot 2024-12-11 at 01 06 16" src="https://github.com/user-attachments/assets/221e4d45-f9fa-4e4b-b4c9-9849c258dbc0"> | <img width="638" alt="Screenshot 2024-12-11 at 01 05 11" src="https://github.com/user-attachments/assets/b2ee1b72-9d1d-4796-b1e3-3c27b82e7fd8"> |
| <img width="682" alt="Screenshot 2024-12-11 at 01 06 11" src="https://github.com/user-attachments/assets/b1930ffc-f9c5-4d2d-ac49-d6eed5418517"> | <img width="638" alt="Screenshot 2024-12-11 at 01 05 03" src="https://github.com/user-attachments/assets/8f046785-b63b-4102-8964-d953ab2b3221"> | 

<!-- Optional section: How to test these changes? -->
<!--
## Testing

-->

<!-- Optional section: What's left to do before it can be merged? -->
<!--
## TODO

What's left to do:

- [ ] ...
- [ ] ...
-->

gtsiolis commented on localstack-core/localstack/cli/localstack.py:
```diff
@@ -908,6 +909,10 @@ def print_profile() -> None:
 
+def print_app() -> None:
+    console.print(
+        f" :globe_with_meridians: [bold]App:[/bold] \033https://app.localstack.cloud\033"
```
**note**: Alternatively, we could use a short URL like https://localstack.dev/app so we can track usage.

alexrashed reviewed (commented):
Thanks a lot for the addition and the great PR description with the screenshots! 🚀 
Some checks on your PR are currently failing, but they are easy to fix:
- "Enforce PR Labels" fails because you _need_ to define the semver for your change by setting one of the `semver: <bump-type>` labels.
- "ci/circleci: preflight" fails because your code is not passing our linting rules.
  - You can easily fix that by using the `make format` and `make lint` commands.
  - You can make sure that this is checked before each of your local commits by installing our pre-commit rules with `make init-precommit`.

Let me know if you need any help :)

gtsiolis commented:
Awesome—thanks for the hints, @alexrashed! 
- Added https://github.com/localstack/localstack/labels/semver%3A%20patch 
- Fixed lint errors in 4f802a1

Could you take another look and let me know what you think about this change? 😃

alexrashed reviewed (commented):
Thanks a lot for addressing the comments! I think it is looking great! 🚀 
But we should clarify the two comments before merging.
Also, please be aware that the changes in the CLI will only take effect with the next release and not immediately after the merge to `master`.
Other than the emulator Docker image, we only publish the CLI on explicit tagged releases.

gtsiolis commented on localstack-core/localstack/cli/localstack.py:
```diff
@@ -909,6 +910,10 @@ def print_profile() -> None:
 
 
+def print_app() -> None:
+    console.print(" :globe_with_meridians: [bold]App:[/bold] \033https://localstack.dev/app\033")
```
Probably copied over for formatting—removed in ff40a2c. 🤖

gtsiolis commented on localstack-core/localstack/cli/localstack.py:
```diff
@@ -909,6 +910,10 @@ def print_profile() -> None:
 
 
+def print_app() -> None:
+    console.print(" :globe_with_meridians: [bold]App:[/bold] \033https://localstack.dev/app\033")
```
Good point—reverted the URL to use our own `app.localstack.cloud` URL.

gtsiolis commented:
Thanks for the insights and taking another look, @alexrashed! 

> [...] changes in the CLI will only take effect with the next release and not immediately after the merge to master.

Sounds good—there'a absolutely no rush to get this in.

Addressed both discussions above.

Since we're here I thought we could consider also dropping the emojis at the start now that it's getting crowded up there to make it a bit more professional—happy to keep the emojis in. What do you think?

| BEFORE  | AFTER (with emojis) | AFTER (with hyphens) |
|--------|--------|--------|
| <img width="682" alt="Screenshot 2024-12-11 at 13 49 29" src="https://github.com/user-attachments/assets/a899d1df-34af-4f51-9bae-a0ce1dab6372"> | <img width="682" alt="Screenshot 2024-12-11 at 13 51 06" src="https://github.com/user-attachments/assets/5e7b715c-a5fa-4957-877e-052934d1e8b4"> | <img width="682" alt="Screenshot 2024-12-11 at 13 48 53" src="https://github.com/user-attachments/assets/53a185c1-39b0-4537-9965-b671cb639d3f"> |

alexrashed reviewed (approved):
Thanks for addressing the comments! And thanks for proactively jumping on these changes! 🦸🏽 
I honestly don't really have an opinion on the emojis. I somewhat like them, but I can see that it's getting a bit too playful. 😉

gtsiolis commented:
Thanks, @alexrashed! ❤️ Removed the emojis as they also become harder to see in dark themes, and updated the PR description to reflect the latest changes.

Merging once the checks pass. 🚀
